src/dune_engine/build.mli

```diff
@@ -212,9 +212,32 @@ val fold_labeled : _ t -> init:'acc -> f:(label -> 'acc -> 'acc) -> 'acc
 
 (** {1 Execution} *)
 
-(** Execute a build description. Returns the result and the set of dynamic
-    dependencies discovered during execution. *)
-val exec : 'a t -> 'a * Dep.Set.t
+module Make_exec (Build_deps : sig
+  val build_deps : Dep.Set.t -> unit Fiber.t
+end) : sig
+  (** Execute a build description. Returns the result and the set of dynamic
+      dependencies discovered during execution. *)
+  val exec : 'a t -> ('a * Dep.Set.t) Fiber.t
+
+  (** Same as [exec] but also builds the static rule dependencies of the build
+      description. *)
+  val build_static_rule_deps_and_exec : 'a t -> ('a * Dep.Set.t) Fiber.t
+end
+
+(** These functions are experimental and potentially unsafe to use. Each usage
+    must be discussed and justified. *)
+module Expert : sig
+  (** If the fiber reads any files, you must ensure they are up to date. *)
+  val fiber : 'a Fiber.t -> 'a t
+
+  (** If the fiber reads any files, you must ensure they are up to date. *)
```

Comment:
I'd mention the caveat of using `Process.run` here instead of file IO. Regular IO should be forbidden in all `Build.t` combinators.

Reply:
Could you clarify (e.g. via suggestions) what you'd like to say about `Process.run`?

Reply:
Caveat is probably the wrong word, as there's nothing wrong with using `Process.run`. It's just that it's quite likely that the caller is likely looking for `Command.run` instead of `Process.run` when generating build rules.

Reply:
Oh, and i should add that `Process.run` should only read the rule dependencies.

Reply:
How about something like this?

If you're thinking of using `Process.run`, check that: (i) you don't in fact need `Command.run`, and (ii) `Process.run` only reads the declared build rule dependencies.